packages/utils/crates/utils-types/src/primitives.rs

```diff
@@ -300,21 +301,25 @@ impl Balance {
     pub fn to_formatted_string(&self) -> String {
         let mut val = self.value.to_string();
 
-        if val.len() > Self::SCALE {
-            let (l, r) = val.split_at(val.len() - Self::SCALE + 1);
-            format!(
-                "{l}.{} {}",
-                &r[..r.len() - (val.len() - Self::SCALE)],
-                self.balance_type
-            )
-        } else if val.len() < Self::SCALE {
-            for _ in 0..(Self::SCALE - val.len() - 1) {
-                val = "0".to_owned() + &val;
+        match val.len().cmp(&Self::SCALE) {
+            Ordering::Greater => {
+                let (l, r) = val.split_at(val.len() - Self::SCALE + 1);
+                format!(
+                    "{l}.{} {}",
+                    &r[..r.len() - (val.len() - Self::SCALE)],
+                    self.balance_type
+                )
+            }
+            Ordering::Less => {
+                for _ in 0..(Self::SCALE - val.len() - 1) {
+                    val = "0".to_owned() + &val;
+                }
+                format!("0.{val} {}", self.balance_type)
```

Comment:
Should be equivalent to:
```suggestion
                format!("0.{empty:0>width$}", empty = &val, width = Self::SCALE - val.len() - 1)
```

Reply:
Nice!

Reply:
Verify it in the rust playground pls though.

Reply:
I did and I had to change it slightly: `format!("0.{empty:0>width$} {currency}",empty = &val,width = Self::SCALE -1,currency = self.balance_type)`